### ISSUE TITLE
set-comprehension false positive on attribute

### ISSUE THREAD

**Issue by ronnodas** [sourcery-ai/sourcery#271]
### Checklist

- [x] I have searched the [Sourcery documentation](https://docs.sourcery.ai/) for the issue, and found nothing
- [x] I have checked there are no open [bugs](https://github.com/sourcery-ai/sourcery/issues?q=label%3Abug) referencing the same bug or problem

### Description
When defining a `set` attribute by using `add` in a loop, `set-comprehension` triggers even when there are conditionals involving the set being added to. The equivalent refactoring does not appear for defining variables. Here is some code sample:
```
class Symmetrizer:
    def __init__(self, pairs):
        self.unordered_pairs = set()
        for x, y in pairs:
            if (y, x) not in self.unordered_pairs:
                self.unordered_pairs.add((x,y))

Symmetrizer({(1, 2), (3, 4), (2, 1)})
```
The suggested refactoring is to replace the `__init__` body by:
```
self.unordered_pairs = {(x, y) for x, y in pairs if (y, x) not in self.unordered_pairs}
```
which raises an `AttributeError`.

### Debug Information

IDE Version: 
PyCharm Professional Edition 2022.2.1

Sourcery Version: 
Sourcery 0.12.7

Operating system and Version:
Linux, Fedora 36

**Comment by ruancomelli** [comment]
Hello, @ronnodas! Thank you for reporting this bug. I was able to reproduce it and will work on a fix. We will let you know once it is fixed.

**Comment by ruancomelli** [comment]
Hi, @ronnodas! A fix for this bug has just been merged. It will be available in the next Sourcery release. Thanks again for reporting this issue!

**Comment by Hellebore** [comment]
Resolved in 0.12.8

**Comment by Hellebore** [comment]
Resolved in 0.12.8
